dask/dataframe/tests/test_io.py

```diff
@@ -820,15 +832,88 @@ def test_to_csv():
         a = dd.from_pandas(df, npartitions)
         with tmpfile('csv') as fn:
             a.to_csv(fn)
-            result = pd.read_csv(fn, index_col=0)
+            result = dd.read_csv(fn).set_index('Unnamed: 0').compute()
```

Comment:
I'd create multiple test functions, with docstrings. Having all tests in one single function makes it much harder to know what is failing when it does.